test/lib/party.js

```diff
@@ -0,0 +1,330 @@
+		(err, res) => {
+		    expect(res).toBe(undefined);
+		    done();	     
+		});
+	    
+	});
+
+	it ('createOrganization should return error fields if req fails', function (done) {
+
+	    nock(url, {
+		reqheaders: headers
+	    }).post(orgPath)
+		.reply(500, {
+		    status: 500,
+		    message: 'An error occurred',
+		    body: 'Internal error detected'
+		});
+	    
+	    var content = {
+		id: '111555999',
+		tradingName: 'AmaneceQueNoEsPoco',
+		href: 'http://exampleuri.com/lack/of/imagination'
+	    };
+	    
+	    orgPartyClient[FUNCTION_MAPPING['mkOrg']](content,
+		(err, res) => {
+		    expect(res).toBe(undefined);
+		    done();	     
+		});
+	    
+	});
+
+	it ('updateOrganization should return error fields if req fails', function (done) {
+
+	    orgP = orgPath + orgId;
+	    var orgPartyClient = partyClient(orgP);
+	    nock(url, {
+		reqheaders: headers
+	    }).patch(orgP)
+		.reply(500, {
+		    status: 500,
+		    message: 'An error occurred',
+		    body: 'Internal error detected'
+		});
+	    
+	    var content = {
+		tradingName: 'LifeOfBrian'
+	    };
+	    
+	    orgPartyClient[FUNCTION_MAPPING['updOrg']](orgId, content, 
+		(err, res) => {
+		    expect(res).toBe(undefined);
+		    done();	     
+		});
+	    
+	});
+
+	it ('getIndividual should return error fields if req fails', function (done) {
+
+	    indP = indPath + indId;
+	    var indPartyClient = partyClient(indP);
+	    
+	    nock(url, {
+		reqheaders: headers
+	    }).get(indP)
+		.reply(500, {
+		    status: 500,
+		    message: 'An error occurred',
+		    body: 'Internal error detected'
+		});
+	    
+	    indPartyClient[FUNCTION_MAPPING['getInd']](indId, 
+		(err, res) => {
+		    expect(res).toBe(undefined);
+		    done();	     
+		});
+	    
+	});
+
+	it ('updateIndividual should return error fields if req fails', function (done) {
+
+	    indP = indPath + indId;
+	    var indPartyClient = partyClient(indP);
+	    
+	    nock(url, {
+		reqheaders: headers
+	    }).patch(indP)
+		.reply(500, {
+		    status: 500,
+		    message: 'An error occurred',
+		    body: 'Internal error detected'
+		});
+
+	    var content = {
+		title: 'Sir lancelot of the Holy Grial'
+	    };
+	    
+	    indPartyClient[FUNCTION_MAPPING['updInd']](indId, content,  
+		(err, res) => {
+		    expect(res).toBe(undefined);
+		    done();	     
+		});
+	    
+	});
+
+	
+    });
+
+    describe('Party API success cases', function () {
+
+	it ('getOrganizations should return the list of all organizations', function (done) {
+
+	    expectedValue = [{ 'id': '111555999',
+			       'href': 'www.example.org/org/1'},
+			     { 'id': '123456789',
+			       'href': 'www.example.org/org/2'}]
+	    nock(url, {
+		reqheaders: headers
+	    }).get(orgPath)
+		.reply(200, expectedValue);
+	    
+	    orgPartyClient[FUNCTION_MAPPING['getOrgs']](
+		(err, res) => {
+		    expect(err).toBeNull();
+		    expect(res.body).toBe(JSON.stringify(expectedValue));
```

Comment:
Validating the string JSON is quite risky. If the stringify method changes the order of any of the attibutes, this is going to fail even if the JSON is the same. Is better to parse the res.body and compare the objects